x/group/server/invariants.go

```diff
@@ -0,0 +1,96 @@
+package server
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/regen-network/regen-ledger/orm"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/regen-network/regen-ledger/types"
+	"github.com/regen-network/regen-ledger/x/group"
+)
+
+func (s serverImpl) RegisterInvariants(ir sdk.InvariantRegistry) {
+	ir.RegisterRoute(group.ModuleName, "Tally-Votes", s.TallyVotesInvariant())
+}
+
+func (s serverImpl) AllInvariants() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		return s.TallyVotesInvariant()(ctx)
+	}
+}
+
+func (s serverImpl) TallyVotesInvariant() sdk.Invariant {
+	return func(sdkCtx sdk.Context) (string, bool) {
+		var msg string
+		var broken bool
+		var curProposal []*group.Proposal
+		ctx := types.Context{Context: sdkCtx}
+		it1, err := s.proposalTable.PrefixScan(ctx, 1, math.MaxUint64)
+		if err != nil {
+			return "start value must be less than end value in iterator", false
+		}
+		curProposalRowID, err := orm.ReadAll(it1, &curProposal)
+		if err != nil {
+			return "cannot read all proposals in current block", false
+		}
+		_ = curProposalRowID
+		var prevProposal []*group.Proposal
+		if ctx.BlockHeight()-1 >= 0 {
+			sdkCtx = sdkCtx.WithBlockHeight(ctx.BlockHeight() - 1)
+		} else {
+			return "Not enough blocks to perform TallyVotesInvariant", false
```

Comment:
I don't believe we should return false in this case, there's just nothing to compare so the invariant is not actually broken. Also we should just verify this at the very beginning of the function so we don't perform `PrefixScan` and `ReadAll` for nothing, ie:
```go
if ctx.BlockHeight() - 1 < 0 {
 return "Not enough blocks to perform TallyVotesInvariant", true
}
```

Reply:
Okay, I'll fix that
Thanks for the info